tests/test_model_signature.py

```diff
@@ -38,7 +38,11 @@ def test_generic_model_signature():
     class Model(BaseModel, Generic[T]):
         a: T
 
-    sig = signature(Model[int])
+    IntModel = Model[int]
+
+    assert IntModel.model_validate({'a': '1'}).a == 1
+
+    sig = signature(IntModel)
```

Comment:
This is really weird! If I don't call `model_validate` the signature is different. It seems like some weirdness with generics. @dmontagu any thoughts?